src/blocks/github-embed/render.php

```diff
@@ -0,0 +1,48 @@
+<?php
+/**
+ * PHP file to use when rendering the block type on the server to show on the front end.
+ *
+ * The following variables are exposed to the file:
+ *     $attributes (array): The block attributes.
+ *     $content (string): The block default content.
+ *     $block (WP_Block): The block instance.
+ *
+ * @see https://github.com/WordPress/gutenberg/blob/trunk/docs/reference-guides/block-api/block-metadata.md#render
+ * @package r3-id-documentation
+ */
+
+
+if ( filter_var( $attributes['code'], FILTER_VALIDATE_URL ) ) {
```

Comment:
if the attributes are split into separate code vs URL attributes for the two different source types, this code here that is checking if it is a URL could be removed and the logic simplified.